lndgrpc/common.py

```diff
@@ -99,19 +99,8 @@ def __init__(self, ip_address='127.0.0.1:10009', network='mainnet', admin=False,
         self._credentials = generate_credentials(cert, macaroon)
         self.ip_address = ip_address
 
-    @property
-    def _ln_stub(self):
-        """Create a ln_stub dynamically to ensure channel freshness
-
-        If we make a call to the Lightning RPC service when the wallet
-        is locked or the server is down we will get back an RPCError with
-        StatusCode.UNAVAILABLE which will make the channel unusable.
-        To ensure the channel is usable we create a new one for each request.
-        """
-        channel = self.grpc_module.secure_channel(
-            self.ip_address, self._credentials, options=[('grpc.max_receive_message_length', 1024*1024*50)]
-        )
-        return lnrpc.LightningStub(channel)
+        channel = self.grpc_module.secure_channel(self.ip_address, self._credentials, options=[('grpc.max_receive_message_length', 1024*1024*50)])
```

Comment:
Is there still an issue getting an `RPCError` if the wallet is locked or server is down? If so I would suggest we add some ability to create a new channel and retry the operation. 

Maybe in the `@handle_rpc_errors` decorator? https://github.com/adrienemery/lnd-grpc-client/blob/master/lndgrpc/errors.py#L13

Reply:
Here are some details on what happened: https://github.com/lightningnetwork/lnd/issues/4031 .

Reply:
I don't understand how decorators work, so I have no opinion on that.

Reply:
In that case my proposal is the following:

1. add an initialization flag (`channel_refresh` maybe?) and if it is `False` then have the stub not create a new channel each time. This way you can get the behaviour you want easily and not break any existing functionality. 

Something like the following:

```python
def __init__(..., channel_refresh=True):
    ...    
    channel = self.grpc_module.secure_channel(self.ip_address, self._credentials, options=[('grpc.max_receive_message_length', 1024*1024*50)])
    self._stub = lnrpc.LightningStub(channel)

@property
def stub()
    if self.channel_refresh:
        # original code
    else:
        return self._stub
```

2. Update the `handle_rpc_erros` decorator so it catches `RPCError`s when the error is `StatusCode.UNAVAILABLE` and then creates a new channel and retries the original request. This would protect against stale channels which was the original intention as well as not create unnecessary channels which fixes the issue you are having. 

Option 1 is more straight forward so I'm happy to merge that option if you go that way. 

I'd rather have option 2 long term though so If you want to do that option 2 after option 1 and are interested in learning about decorators I can point you in the right direction on how to approach it. Or if/when I have time I can implement it and you could review and/or test it?

Thoughts?